src/main/java/dev/jbang/source/TagReader.java

```diff
@@ -153,6 +154,17 @@ protected boolean isDescriptionDeclare(String line) {
 		return line.startsWith(DESCRIPTION_COMMENT_PREFIX);
 	}
 
+	public Optional<String> getDocs() {
```

Comment:
Don't return a `String` but a `ResourceRef`. You can look at methods like `collectFiles()` on how to do that. (Although in this case the method would be much simpler because //DOCS doesn't need to support wildcards. But it does need a resourceRef and a siblingResolver like we do for //SOURCES and //FILES). Don't hesitate to ask me questions if you get stuck!

Reply:
Hi @quintesse thank's a lot for your first feedback and suggestion. It's not clear to me why the `getDoc` method should use a `SiblingResourceResolver`.
I think of this possible implementation:
```java
public Optional<ResourceRef> getDocs(ResourceResolver resolver) {
	return getTags()
		.filter(this::isDocsDeclare)
		.map(s -> s.substring(DOCS_COMMENT_PREFIX.length()))
		.map(resolver::resolve)
		.findFirst();
}
```

Invoking from `ProjectBuilder.createJbangProject` as
```java
prj.setDocs(tagReader.getDocs(ResourceResolver.forResources()).orElse(null));
```

So why should it use a `SiblingResourceResolver` ?

Reply:
> So why should it use a `SiblingResourceResolver` ?

Because a user can do `jbang info docs http://some/repo/or/url/code.java` and if the code.java file has a line like `//DOCS mydocs.html` then the html file has to be retrieved from `http://some/repo/or/url/mydocs.html` and that's what the resource refs are for.

Reply:
Btw if you look at how `TagReader.collectFiles()` is called you'll see it simply wraps `ResourceResolver.forResources()` in a `mew SiblingResolver()`, eg:

```java
ss.addResources(tagReader.collectFiles(resourceRef,
	new SiblingResourceResolver(resourceRef, ResourceResolver.forResources())));
```

So, it's not much more complicated.

Reply:
Oh, I see there might also be somewhat of a miscommunication: I'm not saying the *type* of the argument should be a `SiblingResourceResolver`, in fact it should be a plain `ResourceResolver`.

Explanation: the way the resolver will be used is always to resolve siblings (when using relative paths the docs are "siblings" of the main source file, as are source and resource files). You can see this in the implementation of `collectSources()` and `collectFiles()`, the parameter is `ResourceResolver siblingResolver`, the "sibling" is only in the name.

Reply:
Sure, although I wouldn't worry about that too much right now. You have to think that possibly the documents aren't files that can be shown directly in the browser. Think of .md and .adoc files that have to be transformed first, which means having them locally available is most likely required anyway.

Still I could imagine the use for something like a `LazyResourceRef` that would delay the resolving part to the moment that somebody actually wants to access the contents of the resource. Then as long as the code that displays the docs never asks for the contents (because it simply passes the url to a browser)  it would never get downloaded. But that could be a future improvement, so again, don't worry about it now.

Reply:
Fixed by dcee2c8f3913144b4027efc8e94d84bd8e2d1ce1